src/connection_string.ts

```diff
@@ -98,14 +109,15 @@ export function resolveSRVRecord(options: MongoOptions, callback: Callback<HostA
 
         const txtRecordOptions = new URLSearchParams(record[0].join(''));
         const txtRecordOptionKeys = [...txtRecordOptions.keys()];
-        if (txtRecordOptionKeys.some(key => key !== 'authSource' && key !== 'replicaSet')) {
+        if (txtRecordOptionKeys.some(key => !VALID_TXT_RECORDS.includes(key))) {
           return callback(
-            new MongoParseError('Text record must only set `authSource` or `replicaSet`')
+            new MongoParseError(`Text record must only set one of: ${VALID_TXT_RECORDS.join(', ')}`)
```

Comment:
```suggestion
            new MongoParseError(`Text record may only set any of: ${VALID_TXT_RECORDS.join(', ')}`)
```

(Otherwise this sounds like it's saying that at most one option can be set -- feel free to adjust the text, of course, or feel free to ignore, as obviously this is a fairly minor thing, I just noticed it :))

Reply:
Updated to suggestion.